src/ui/buttons/props.js

```diff
@@ -430,7 +430,7 @@ export type ApplePaySessionConfigRequest = (
 
 export type ButtonMessage = {|
   amount?: number,
-  offer?: $ReadOnlyArray<$Values<typeof MESSAGE_OFFER>>,
+  offer?: $Values<typeof MESSAGE_OFFER> | string,
```

Comment:
if this is going to be a comma separated string, should the type just be `string`?

Reply:
that is a good call, as the `join` statement makes the output a generic string, instead of our strict flow type.

Reply:
I think this may be meant to be `$ReadOnlyArray<$Values<typeof MESSAGE_OFFER>> | string` since we want to accept the input as both and array or comma-delimited string. The merchant input will be `["a", "b"]` vs `"a,b"` for better ergonomics. We convert it internally to a comma-delimited string so that we pass it to the endpoint as a string to get around the `zoid` `dotify` encoding issue.

Reply:
Oh apologies. I misspoke and got my wires crossed. `ButtonMessage` is the output of the function and not the input. :sweat_smile: My comment above can be disregarded.